Add Alibaba Cloud Object Storage Service (OSS) backend for etcd-backu… (#2065)
coreos/etcd-operator: doc/design/oss_backup.md

# Backups using Alibaba Cloud Object Storage Service (OSS)

## Cluster configured with OSS backup

> To create a backup in OSS, set `backup.storageType` to `"OSS"`, supply the path (in the format "<oss-bucket-name>/<path-to-backup-object>") in `oss.path` and provide the Kubernetes secret storing the Alibaba Cloud account credentials to `oss.ossSecret`.  The secret must exist prior to backup creation. Etcd backup operator will create the bucket and object if not found. The field `oss.endpoint` means the target OSS service endpoint where backs up the data to, by default, "http://oss-cn-hangzhou.aliyuncs.com" will be used, if you want to back up the data to other regions, please follow [here](https://www.alibabacloud.com/help/doc-detail/31837.htm).

Reviewer: Some rephrasing and formatting nits:
- Change formatting for `<oss-bucket-name>/<path-to-backup-object>` to show angle brackets `<>` correctly on Github flavored markdown.
- Grammar fix: 
    - ```The field `oss.endpoint` means the target OSS service endpoint where backs up the data to, by default...``` => ```The field `oss.endpoint` is the target OSS service endpoint where the data is backed up. By default...```
- `please follow [here]` ==> `please specify another endpoint from [the list of region endpoints]`
```suggestion
To create a backup in OSS, set `backup.storageType` to `"OSS"`, supply the path (in the format `<oss-bucket-name>/<path-to-backup-object>`) in `oss.path` and provide the Kubernetes secret storing the Alibaba Cloud account credentials to `oss.ossSecret`.  The secret must exist prior to backup creation. Etcd backup operator will create the bucket and object if not found. The field `oss.endpoint` is the target OSS service endpoint where the data is backed up. By default, `http://oss-cn-hangzhou.aliyuncs.com` will be used. If you want to back up the data to other regions, please specify another endpoint from [the list of region endpoints](https://www.alibabacloud.com/help/doc-detail/31837.htm).
```

> To create a cluster with backups to OSS, set `backup.storageType` to `"OSS"`, supply the path (in the format "<oss-bucket-name>/<path-to-backup-object>") in `oss.path` and provide the Kubernetes secret storing the Alibaba Cloud account credentials to `oss.ossSecret`.  The secret must exist prior to backup creation. Etcd backup operator will create the bucket and object if not found. The field `oss.endpoint` means the target OSS service endpoint where backs up the data to, by default, "http://oss-cn-hangzhou.aliyuncs.com" will be used, if you want to back up the data to other regions, please follow [here](https://www.alibabacloud.com/help/doc-detail/31837.htm).

Reviewer 2: maybe `To create a etcd cluster backup in OSS`

> An example cluster manifest would look like:

Reviewer 2: An example cluster manifest would look like:-》 An example backup manifest would look like:

```yaml
apiVersion: etcd.database.coreos.com/v1beta2
kind: EtcdBackup
metadata:
  name: etcd-cluster-with-oss-backup
  namespace: my-namespace
spec:
  backupPolicy:
    ...
  etcdEndpoints:

>     ...

Reviewer: Let's keep this example consistent with the examples for other backup object types so it works as is:
```suggestion
    - "http://example-etcd-cluster-client:2379"
```
Also remove lines 17-18 since we don't need periodic backup for this example.

  storageType: OSS
  oss:
    endpoint: http://oss-cn-hangzhou.aliyuncs.com
    ossSecret: my-oss-credentials
    path: my-etcd-backups-bucket/etcd.backup
```

### In Detail:

- `"ossSecret"` represents the name of the Kubernetes secret object that stores the credentials needed for Alibaba Cloud authorization, namely an authorization token.

>   The Kubernetes secret manifest looks like either:

Reviewer: ```suggestion
  The Kubernetes secret manifest must have the following format:
```
